### PR TITLE
Housekeeping/fix components 3.7.0 rc.1

### PR DESCRIPTION
**Pull request by tatjana-gartner2**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/housekeeping/fix-components-3.7.0-rc.1

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves [IssueNumber]

### CODE REVIEW

File: packages/components/src/utils/validation/partials/validatePartialUsage.spec.ts [hunk 1]
```diff
@@ -46,16 +46,7 @@ describe('validatePartialUsage()', () => {
...
-    ['production', 'test'],
-    ['production', 'production'],
-  ])(
-    'should call validateGetInitialStylesUsage(), validateGetFontFaceStylesheetUsage() and validateGetFontLinksUsage() for ROLLUP_REPLACE_IS_STAGING: "%s" and process.env.NODE_ENV: "%s"',
-    (rollupReplaceIsStaging, nodeEnv) => {
-      // @ts-ignore
-      ROLLUP_REPLACE_IS_STAGING = rollupReplaceIsStaging;
-      process.env = { ...originalEnv, NODE_ENV: nodeEnv };
-
+  it('should call validateGetInitialStylesUsage(), validateGetFontFaceStylesheetUsage() and validateGetFontLinksUsage()"', () => {
```
Comment (denyo) [inline packages/components/src/utils/validation/partials/validatePartialUsage.spec.ts:49]:
```suggestion
  it('should call validateGetInitialStylesUsage(), validateGetFontFaceStylesheetUsage() and validateGetFontLinksUsage()', () => {
```

File: packages/components-js/tests/e2e/puppeteer/helpers/puppeteer-helper.ts [hunk 2]
```diff
@@ -43,6 +44,8 @@ export const setContentWithDesignSystem = async (page: Page, content: string, op
         <link rel="stylesheet" href="http://localhost:3001/styles/font-face.min.css">
         <link rel="stylesheet" href="assets/styles.css">
         ${options.injectIntoHead}
+        ${getFontFaceStylesheet()}
+        ${getInitialStyles()}
```
Comment (denyo) [inline packages/components-js/tests/e2e/puppeteer/helpers/puppeteer-helper.ts:48]:
```suggestion
        ${getInitialStyles()}
        ${getFontFaceStylesheet().replace(/https:\/\/cdn\.ui\.porsche\.com\/porsche-design-system/g, 'http://localhost:3001')}
        ${options.injectIntoHead}
```

File: packages/components-js/public/index.html [hunk 3]
```diff
@@ -15,6 +15,7 @@
...
     <!--PLACEHOLDER_PORSCHE_DESIGN_SYSTEM_FONT_LATIN--><link rel=preload href=http://localhost:3001/fonts/porsche-next-w-la-regular.min.b8f1c203bd2ac5334247025236ba8d86.woff2 as=font type=font/woff2 crossorigin><link rel=preload href=http://localhost:3001/fonts/porsche-next-w-la-semi-bold.min.b5f6fca7225aa73578f65320bcf79b45.woff2 as=font type=font/woff2 crossorigin><link rel=preload href=http://localhost:3001/fonts/porsche-next-w-la-bold.min.0fbdc6d73f04137fff6e4169e8c30e2c.woff2 as=font type=font/woff2 crossorigin>
     <link rel="stylesheet" href="assets/styles.css" />
+    <link rel="stylesheet" href="http://localhost:3001/styles/font-face.min.css" />
```
Comment (denyo) [inline packages/components-js/public/index.html:18]: Duplicate of L14.
```suggestion
```

File: packages/components/src/index.html [hunk 4]
```diff
@@ -10,6 +10,7 @@
     <link rel="stylesheet" type="text/css" href="http://localhost:3001/styles/font-face.min.css" />
     <script nomodule src="build/porsche-design-system.js"></script>
     <script type="module" src="build/porsche-design-system.esm.js"></script>
+    <style data-pds-initial-styles></style>
     <style>
```
Comment (denyo) [inline packages/components/src/index.html:14]: Why not use the `style` tag that is already there? 
```suggestion
    <!-- to pass partial validation via validateGetInitialStylesUsage() -->
    <style data-pds-initial-styles>
```

File: packages/components-angular/karma.conf.js [hunk 5]
```diff
@@ -1,11 +1,33 @@
...
+const transformIndexHtml = require('./scripts/transformIndexHtml');
 
 if (process.env.CI) {
   // using chrome that comes with playwright
   process.env.CHROME_BIN = require('@playwright/test').chromium.executablePath();
 }
 
+const injectPartialsIntoKarmaContextHtml = () => {
+  const packagePath = path.resolve(require.resolve('@angular-devkit/build-angular'), '..');
+  const [contextHtml] = globSync(packagePath + '/**/karma-context.html');
```
Comment (denyo) [inline packages/components-angular/karma.conf.js:15]: Calling this `filePath` would be more consisting with the rest of the code.

File: packages/components-angular/package.json [hunk 6]
```diff
@@ -38,8 +38,9 @@
     "@angular/platform-browser": "^15.2.9",
     "@angular/platform-browser-dynamic": "^15.2.9",
     "@angular/router": "^15.2.9",
-    "@porsche-design-system/components-angular": "3.7.0-rc.1",
+    "@porsche-design-system/components-angular": "3.7.0-rc.2",
     "angular-imask": "^6.6.3",
+    "glob": "^10.3.10",
```
Comment (denyo) [inline packages/components-angular/package.json:43]: Should be a `devDependency`?

File: packages/storefront/src/components/PartialDocs.vue [hunk 7]
```diff
@@ -68,6 +68,14 @@ yarn add --dev @angular-builders/custom-webpack
     "options": {
       "outputPath": "dist/components-angular",
 +     "indexTransform": "./scripts/transformIndexHtml.ts",
+    }
+  }
+  "serve": {
+-   "builder": "@angular-devkit/build-angular:dev-server",
++   "builder": "@angular-builders/custom-webpack:dev-server",
+    "options": {
++     "browserTarget": "demo-app:build"
```
Comment (denyo) [inline packages/storefront/src/components/PartialDocs.vue:77]: Pretty sure that this won't work for anyone but us.

File: packages/storefront/src/components/PartialDocs.vue [hunk 8]
```diff
@@ -77,7 +85,35 @@ export default (targetOptions: TargetOptions, indexHtml: string): string => {
   ${angularPartials}
 
   return indexHtml.replace(/<\\/${this.location}>/, \`\${partialContent}$&\`);
-};`;
+};
+
+yarn add --dev glob
+
+<!-- karma.conf -->
```
Comment (denyo) [inline packages/storefront/src/components/PartialDocs.vue:92]: Why not put this on a dedicated page in the onboarding similar to other frameworks? https://designsystem.porsche.com/housekeeping/fix-components-3.7.0-rc.1/developing/react/testing
Currently this is just duplicated a bunch of times in each partial demo without any context and little value.

File: packages/storefront/src/pages/developing/angular/testing.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+## Testing with Karma
+
+To enable the [`getInitialStyles()`](partials/initial-styles) partial within your Karma tests, add the following code to
+your `karma.conf`.
+
+```tsx
+// yarn add --dev glob
+
+<!-- karma.conf -->
```
Comment (denyo) [inline packages/storefront/src/pages/developing/angular/testing.md:13]: File extension?

File: packages/storefront/src/pages/developing/angular/testing.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
+# Angular
+
+<TableOfContents></TableOfContents>
+
+## Testing with Karma
+
+To enable the [`getInitialStyles()`](partials/initial-styles) partial within your Karma tests, add the following code to
+your `karma.conf`.
```
Comment (denyo) [inline packages/storefront/src/pages/developing/angular/testing.md:8]: File extension?

File: packages/storefront/src/pages/developing/angular/testing.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+**Jest** uses **jsdom** and supports ShadowDOM since Version 12.2.0.  
+However, it doesn't support JavaScript modules as described in this
+[issue](https://github.com/jsdom/jsdom/issues/2475).  
+Also, it doesn't support `CSSStyleSheet.replace()`, `Intersection Observer`, `Element.prototype.scrollTo` and others.
+
+As a workaround we provide a polyfill as part of the `@porsche-design-system/components-angular` package.
+
+To apply the polyfill, simply import it in your **setupTest.{js|ts}** file.
+
+**Note:** If your test includes Porsche Design System components, make sure to import the PorscheDesignSystemModule in
```
Comment (o-hook) [inline packages/storefront/src/pages/developing/angular/testing.md:16]: Maybe better:
"_make sure that you import the PorscheDesignSystemModule into
each module in which you want to use the components._"

**Review by o-hook — DISMISSED** [review]

**Review by o-hook — APPROVED** [review]

**Pull request merged**